File: 00961. N-Repeated Element in Size 2N Array/solution.cpp
/**
 * Problem: 961. N-Repeated Element in Size 2N Array
 *
 * Difficulty: Easy
 *
 * Language: C++
 *
 * Performance: Runtime - 0 ms (Beats 100%)
 */

class Solution {
public:
    int repeatedNTimes(vector<int>& nums) {
        // Create an unordered set to track encountered numbers
        unordered_set<int> s;

        // Iterate through each element in the input vector
        for (int x : nums) {
            // Attempt to insert the current element into the unordered set
            // The insert method returns a pair (iterator, bool) where the bool
            // indicates success If the element already exists in the set
            // (second is false), it is the repeated element we're looking for
            if (!s.insert(x).second)
                return x;
        }

        // Return -1 if no repeated element is found (should not happen per
        // problem constraints)
        return -1;
    }
};

File: 00961. N-Repeated Element in Size 2N Array/solution.cs
/**
 * Problem: 961. N-Repeated Element in Size 2N Array
 *
 * Difficulty: Easy
 *
 * Language: C#
 *
 * Performance: Runtime - 0 ms (Beats 100%)
 */

public class Solution
{
    public int RepeatedNTimes(int[] nums)
    {
        // Calculate n based on the array length (array size is 2n)
        var n = nums.Length / 2;

        // Create a dictionary to store the frequency of each number
        var dict = new Dictionary<int, int>();
        
        // Iterate through each number in the input array
        foreach (var number in nums)
        {
            // Check if the number already exists in the dictionary
            if (dict.ContainsKey(number))
            {
                // If the number is already in the dictionary, it's the repeated element
                // No need to count frequencies - return immediately when first duplicate is found
                return number;
            }
            // If the number is not in the dictionary, add it with count 1
            else dict.Add(number, 1);
        }

        // Return 0 if no repeated element is found (should not happen per problem constraints)
        return 0;
    }
}